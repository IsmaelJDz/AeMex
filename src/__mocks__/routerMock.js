import React, { Children } from "react";
import { BrowserRouter } from "react-router-dom";

const routerMock = props => <BrowserRouter>{Children}</BrowserRouter>;

export default routerMock;
