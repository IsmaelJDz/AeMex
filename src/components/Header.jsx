import React, { Fragment } from "react";

// ================================
// Style Sheet Component
// =================================
import "../styles/components/Header.scss";

//===================================
//  Resources Img
//===================================
import LogoAeroMex from "../resources/img/logoAero.svg";
import LogoAeroMexResponsive from "../resources/img/ResponsiveLogo.svg";
import Pictogram from "../resources/img/logoAeroTwo.svg";
import Language from "../resources/img/mexFlag.svg";

// ===================================
// State less functional component
// ===================================

const Header = () => {
  return (
    <Fragment>
      <div className="Header--main">
        <div className="Header--images">
          <div className="Header--images-logoResponsive">
            <img
              src={LogoAeroMexResponsive}
              className="Header--responsive"
              alt="Logo"
              width="35px"
            />
          </div>
          <div className="Header--images-logo">
            <img
              src={LogoAeroMex}
              className="Header--mainLogo"
              alt="Logo"
              width="140px"
            />
          </div>
          <div className="Header--images-logoPic">
            <img
              src={Pictogram}
              className="Header--images__secondaryLogo"
              alt="Pictogram"
              width="25px"
            />
          </div>
        </div>

        <div className="Header--reservation">
          <h4 className="Header--reservation-text">RESERVA</h4>
        </div>

        <div className="Header--menu_main">
          <nav>
            <ul>
              <li>Reserva</li>
              <li>Tu viaje</li>
              <li>Check-In</li>
              <li>Upgrade</li>
              <li>Club Premier</li>
            </ul>
          </nav>
        </div>
        <div className="Header--menu_secondary">
          <div className="Header--menu_secondary-travel">
            <ul>
              <li>Promociones</li>
              <li>
                <span className="Header--menu_secondary-activeLink">
                  Rastrea un vuelo
                </span>
              </li>
              <li>Destinos</li>
              <li>Más</li>
            </ul>
          </div>
          <div className="Header--menu_secondary-profile">
            <span className="icon-envelop"></span>
            <img
              src={Language}
              alt="idioma"
              width="25px"
              className="Header--menu_secondary-imgLanguage"
            />
            <a
              to="/"
              className="Header--menu_secondary-profile_login"
              alt="login"
            >
              Iniciar sesión
            </a>
          </div>
          <div className="Header--menu-tablet">
            <div className="Header--menu-flag">
              <span className="icon-envelop"></span>
            </div>
            <div className="Header--menu-hamburguer">
              <span className="icon-menu"></span>
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
};

export default Header;
