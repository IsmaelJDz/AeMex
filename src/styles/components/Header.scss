$base-color: #0b2343;
$secondary-color: #007cc2;
$letter-color: #fff;
$font-main: "NeueHaasUnicaPro";

.Header--main {
  align-items: center;
  background-color: $base-color;
  display: flex;
  flex-wrap: wrap;
}

.Header--images {
  margin-left: 15px;
  &__secondaryLogo {
    margin-left: 10px;
  }
}

.Header--menu_main {
  list-style: none;
  color: $letter-color;
  & nav {
    ul {
      margin-right: 10px;
      display: flex;
      list-style: none;
      li {
        margin: 10px;
      }
    }
  }
}

.Header--menu_secondary {
  list-style: none;
  color: $letter-color;
  font-family: $font-main;
  display: flex;
  flex-wrap: wrap;
  font-size: 12px;
  margin-left: auto;
  &-travel ul {
    list-style: none;
    display: flex;
    margin-right: 10px;
    li {
      margin: 10px;
    }
  }
  &-profile {
    width: 200px;
    display: flex;
    justify-content: space-around;
    align-content: center;
    align-items: center;
    border-left: 1px solid #777777;
  }

  &-profile_flag img {
    border-radius: 100%;
  }

  &-profile_login span {
    color: $letter-color;
    margin-right: 30px;
  }
}
