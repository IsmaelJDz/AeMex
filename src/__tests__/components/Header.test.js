import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import routerMock from "../../__mocks__/routerMock";
import Header from "../../components/Header";

configure({ adapter: new Adapter() });

describe("<Header />", () => {
  test("Renderizar el header", () => {
    const header = shallow(
        <Header />
    );
    expect(header.length).toEqual(1);
  });
});

describe("Header SnapShot", () => {
  test("comprobar el SnapShot de Header", () => {
    const header = renderer.create(
      <routerMock>
        <Header />
      </routerMock>
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
