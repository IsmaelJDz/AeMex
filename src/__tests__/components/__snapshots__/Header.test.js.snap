// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header SnapShot comprobar el SnapShot de Header 1`] = `
<routerMock>
  <div
    className="Header--main"
  >
    <div
      className="Header--images"
    >
      <div
        className="Header--images-logoResponsive"
      >
        <img
          alt="Logo"
          className="Header--responsive"
          src="ResponsiveLogo.svg"
          width="35px"
        />
      </div>
      <div
        className="Header--images-logo"
      >
        <img
          alt="Logo"
          className="Header--mainLogo"
          src="logoAero.svg"
          width="140px"
        />
      </div>
      <div
        className="Header--images-logoPic"
      >
        <img
          alt="Pictogram"
          className="Header--images__secondaryLogo"
          src="logoAeroTwo.svg"
          width="25px"
        />
      </div>
    </div>
    <div
      className="Header--reservation"
    >
      <h4
        className="Header--reservation-text"
      >
        RESERVA
      </h4>
    </div>
    <div
      className="Header--menu_main"
    >
      <nav>
        <ul>
          <li>
            Reserva
          </li>
          <li>
            Tu viaje
          </li>
          <li>
            Check-In
          </li>
          <li>
            Upgrade
          </li>
          <li>
            Club Premier
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="Header--menu_secondary"
    >
      <div
        className="Header--menu_secondary-travel"
      >
        <ul>
          <li>
            Promociones
          </li>
          <li>
            <span
              className="Header--menu_secondary-activeLink"
            >
              Rastrea un vuelo
            </span>
          </li>
          <li>
            Destinos
          </li>
          <li>
            Más
          </li>
        </ul>
      </div>
      <div
        className="Header--menu_secondary-profile"
      >
        <span
          className="icon-envelop"
        />
        <img
          alt="idioma"
          className="Header--menu_secondary-imgLanguage"
          src="mexFlag.svg"
          width="25px"
        />
        <a
          alt="login"
          className="Header--menu_secondary-profile_login"
          to="/"
        >
          Iniciar sesión
        </a>
      </div>
      <div
        className="Header--menu-tablet"
      >
        <div
          className="Header--menu-flag"
        >
          <span
            className="icon-envelop"
          />
        </div>
        <div
          className="Header--menu-hamburguer"
        >
          <span
            className="icon-menu"
          />
        </div>
      </div>
    </div>
  </div>
</routerMock>
`;
